week1/intro.cpp

```diff
@@ -0,0 +1,27 @@
+class Foo {
+public:
+    std::mutex two,three;
+    Foo() {
+        two.lock();
+        three.lock();    
+    }
+
+    void first(function<void()> printFirst) {
+        // printFirst() outputs "first". Do not change or remove this line.
+        printFirst();
+        two.unlock();
```

Comment:
Also, does unlocking a mutex notify other threads that are waiting on it?

Reply:
yes